crates/ra_hir/src/expr.rs

```diff
@@ -794,7 +789,33 @@ impl ExprCollector {
             ast::ExprKind::Label(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::IndexExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::RangeExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
-            ast::ExprKind::MacroCall(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
+            ast::ExprKind::MacroCall(e) => {
+                // very hacky.FIXME change to use the macro resolution
+                let name = e
+                    .path()
+                    .and_then(Path::from_ast)
+                    .and_then(|path| path.expand_macro_expr())
+                    .unwrap_or_else(Name::missing);
+
+                if let Some(call_id) = self.resolver.resolve_macro_call(self.db, &name, e) {
+                    let file_id: HirFileId = call_id.into();
+
+                    log::debug!(
+                        "expanded macro ast {}",
+                        self.db.hir_parse(file_id).syntax().debug_dump()
+                    );
+
+                    self.db
+                        .hir_parse(file_id)
+                        .syntax()
+                        .descendants()
```

Comment:
So the last big bit here is to use "parse macro into expression" bit which is being implemented by @edwin0cheng in https://github.com/rust-analyzer/rust-analyzer/pull/1148.

Sorry for not realizing from the outset that we have a couple of puzzle pieces missing here to "just" implement this :)

Reply:
That PR is merged, so I think this could be pushed over the finish line?

Reply:
We kinda need the sourceid part as the macro_query expands everything to a source file. Or I can open a new pr which does macro_expansion diffrently and expand it some expressions

Reply:
Yeah, I think we need to introduce `expand_macro_to_expr` query. It probably shouldn't even be a query, just a helper function

```
fn expand_macro_to_expr(db: &impl HirDatabase, macro_call: MacroCallId) -> Option<TreeArc<ast::Expr>>
```